staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.pb.go

```diff
@@ -9653,7 +9653,7 @@ func init() {
 func init() { proto.RegisterFile("api.proto", fileDescriptor_00212fb1f9d3bf1c) }
 
 var fileDescriptor_00212fb1f9d3bf1c = []byte{
-	// 6539 bytes of a gzipped FileDescriptorProto
+	// 6514 bytes of a gzipped FileDescriptorProto
```

Comment:
*raises eyebrow*

was this stale?

Reply:
No.  Another rabbithole.  It's encoded gzipped data.  Apparently gzip is not deterministic (so makes a GREAT choice for this embedded data /s) and this blob encodes the go_package.  If you decode and diff this blob the delta is:

```diff
@@ -4684,7 +4684,7 @@
         }
       ],
       "options": {
    -    "go_package": "k8s.io/cri-api/pkg/apis/runtime/v1"
    +    "go_package": "v1"
       },
       "syntax": "proto3"
     }
 ```

I put this in the commit message for "Move update-generated-runtime into common script".

What I can't tell is if this really matters.

Reply:
`"go_package": "v1"` looks wrong... isn't that supposed to be the full package?

Reply:
Yeah - I don't know how this is used.  I am not sure on it and need to figure it out. 

I ran the same decode on ALL the pb.go files and only one other ("dra") sets this option, and it sets it to "dra".

PB code documents this field as:

```
    // Sets the Go package where structs generated from this .proto will be
    // placed. If omitted, the Go package will be derived from the following:
    //   - The basename of the package import path, if provided.                                                
    //   - Otherwise, the package statement in the .proto file, if present.
    //   - Otherwise, the basename of the .proto file, without extension.
```

Which, to my read, sounds like "v1" is correct and "k8s.io/cri-api/pkg/apis/runtime/v1" was always wrong.

Your opinion?

Reply:
but... it doesn't match the proto file, right?

https://github.com/kubernetes/kubernetes/blob/a66aad2d80dacc70025f95a8f97d2549ebd3208c/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L21

https://github.com/kubernetes/kubernetes/pull/106630 pointed at https://developers.google.com/protocol-buffers/docs/reference/go-generated#package which said:

> The Go import path is locally specified in a .proto file by declaring a go_package option with the full import path of the Go package. Example usage:
>
> `option go_package = "example.com/project/protos/fizz";`

Reply:
Indeed.  I have reached out to Go libs team (but they are in EU,  so it's weekend).  It does seem like a full path is more common.

Several of these kubelet gRPC APIs don't specify this option at all, and it's not clear to me what impact that has.

The problem is that along the way the directory and the package get concatenated  so we end up with staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/k8s.io/cri-api/pkg/apis/runtime/v1/api.pb.go

Reply:
I can reformulate how protoc is called and remove this diff.  Will be in my next push